compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -1300,12 +1300,16 @@ fn start_executing_work<B: ExtraBackendMethods>(
         let mut main_thread_state = MainThreadState::Idle;
 
         // How many LLVM worker threads are running while holding a Token. This
-        // *excludes* the LLVM worker thread that the main thread is lending a
-        // Token to (when the main thread is in the `Lending` state).
-        // In other words, the number of LLVM threads is actually equal to
-        // `running + if main_thread_state == Lending { 1 } else { 0 }`.
+        // *excludes* any that the main thread is lending a Token to.
         let mut running_with_own_token = 0;
 
+        // How many LLVM worker threads are running in total. This *includes*
+        // any that the main thread is lending a Token to.
+        let running_with_any_token = |main_thread_state, running_with_own_token| {
+            running_with_own_token
+                + if main_thread_state == MainThreadState::Lending { 1 } else { 0 }
+        };
```

Comment:
I think this function should return a bool. It is always compared against 0 and "any" is an operation which normally returns a bool.

Reply:
I considered that, and I felt that the `running_with_own_token`/`running_with_any_token` distinction was better because it shows they are closely related. But I could be convinced otherwise.